src/cargo/core/resolver/types.rs

```diff
@@ -299,48 +305,45 @@ impl Ord for DepsFrame {
     fn cmp(&self, other: &DepsFrame) -> Ordering {
         self.just_for_error_messages
             .cmp(&other.just_for_error_messages)
-            .then_with(||
-            // the frame with the sibling that has the least number of candidates
-            // needs to get bubbled up to the top of the heap we use below, so
-            // reverse comparison here.
-            self.min_candidates().cmp(&other.min_candidates()).reverse())
+            .reverse()
+            .then_with(|| self.min_candidates().cmp(&other.min_candidates()))
     }
 }
 
-/// Note that a `BinaryHeap` is used for the remaining dependencies that need
-/// activation. This heap is sorted such that the "largest value" is the most
-/// constrained dependency, or the one with the least candidates.
+/// Note that a `OrdSet` is used for the remaining dependencies that need
+/// activation. This set is sorted by how many candidates each dependency has.
 ///
 /// This helps us get through super constrained portions of the dependency
 /// graph quickly and hopefully lock down what later larger dependencies can
 /// use (those with more candidates).
 #[derive(Clone)]
-pub struct RemainingDeps(BinaryHeap<DepsFrame>);
+pub struct RemainingDeps(usize, im_rc::OrdSet<(DepsFrame, usize)>);
```

Comment:
I've need trying to find good names for the two `usize` fields, and failing.  Can you think of good names and label the fields in code?  It think it would help understanding the code.